pallets/phala/src/fat.rs

```diff
@@ -218,17 +265,63 @@ pub mod pallet {
 	where
 		T: crate::mq::Config + crate::registry::Config,
 	{
-		pub fn on_contract_message_received(
-			message: DecodedMessage<ContractRegistryEvent>,
+		pub fn on_cluster_message_received(
+			message: DecodedMessage<ClusterRegistryEvent>,
 		) -> DispatchResult {
 			ensure!(
 				message.sender == MessageOrigin::Gatekeeper,
 				Error::<T>::InvalidSender
 			);
+			match message.payload {
+				ClusterRegistryEvent::PubkeyAvailable { cluster, pubkey } => {
+					registry::ClusterKeys::<T>::insert(&cluster, &pubkey);
+					Self::deposit_event(Event::ClusterPubkeyAvailable { cluster, pubkey });
+				}
+			}
+			Ok(())
+		}
+
+		pub fn on_contract_message_received(
+			message: DecodedMessage<ContractRegistryEvent>,
+		) -> DispatchResult {
+			let cluster = match message.sender {
+				MessageOrigin::Cluster(cluster) => cluster,
+				_ => return Err(Error::<T>::InvalidSender.into()),
+			};
 			match message.payload {
 				ContractRegistryEvent::PubkeyAvailable { contract, pubkey } => {
-					registry::ContractKeys::<T>::insert(contract, pubkey);
-					Self::deposit_event(Event::PubkeyAvailable { contract, pubkey });
+					registry::ContractKeys::<T>::insert(&contract, &pubkey);
```

Comment:
Is it ECDH key?

Reply:
For now, the `ContractPubKey` is used as `EcdhPubKey`.